### PR TITLE
Sets and enums

### PR DESCRIPTION
**Pull request by osheroff**
@vanchi-zendesk I had been avoiding sets and enum support for awhile, thinking it was hard.  It wasn't, particularly...

### CODE REVIEW

File: src/main/java/com/zendesk/maxwell/schema/SchemaCapturer.java [hunk 1]
```diff
@@ -110,8 +114,26 @@ private void captureTable(Table t) throws SQLException {
...
+			}
+
+			t.getColumnList().add(ColumnDef.build(t.getName(), colName, colEnc, colType, colPos, colSigned, enumValues));
 			i++;
 		}
 	}
+
+	private static String[] extractEnumValues(String expandedType) {
+		String[] enumValues;
+		Matcher matcher = Pattern.compile("(enum|set)\\((.*)\\)").matcher(expandedType);
```
Comment (vanchi-zendesk) [inline src/main/java/com/zendesk/maxwell/schema/SchemaCapturer.java]: what's up with the double slash?

File: src/main/java/com/zendesk/maxwell/schema/columndef/SetColumnDef.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+		return asList(value);
+	}
+
+	private ArrayList<String> asList(Object value) {
+		ArrayList<String> values = new ArrayList<>();
+		long v = (Long) value;
+		for(int i = 0; i < enumValues.length; i++) {
+			if ( ((v >> i) & 1) == 1 ) {
+				values.add(enumValues[i]);
+			}
```
Comment (vanchi-zendesk) [inline src/main/java/com/zendesk/maxwell/schema/columndef/SetColumnDef.java]: Oh dear god. Why does mysql have this? Do we ever use it?

**Comment by vanchi-zendesk** [comment]
:+1:

File: src/main/java/com/zendesk/maxwell/schema/columndef/SetColumnDef.java (hunk 2 above)
Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/schema/columndef/SetColumnDef.java]: sets are useful!  Zendesk doesn't tend to use them...

File: src/main/java/com/zendesk/maxwell/schema/SchemaCapturer.java (hunk 1 above)
Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/schema/SchemaCapturer.java]: it allows you to pass the backslash into the regexp engine to escape the parens: the string reads literally `(enum|set)\((.*)\)`

Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/schema/SchemaCapturer.java]: that is to say, a backslash in a java string will normally escape the next character.

**Pull request merged**